packages/unmock-core/src/service/service.ts

```diff
@@ -48,17 +134,62 @@ export class Service {
     return true;
   }
 
-  // TODO
-  // private verifyPathAndMethod(method: HTTPMethod, path: string) {
-  //   if (path === DEFAULT_ENDPOINT) {
-  //     if (method === DEFAULT_REST_METHOD) {
-  //       return this.oasSchema.paths !== undefined;
-  //     } else {
-  //       // just make sure we have some entry that matches the method in that level
-  //       return getAtLevel(this.schema, 1, (k, _) => k === method).length > 0;
-  //     }
-  //   } else {
-  //     // make sure the combination exists
-  //   }
-  // }
+  private __complementPathWithRegexp(newPath: string, oldPath: string) {
+    // Complements oldPath by adding an unmock extension for regex matching the path.
+    const newPathRegex = XRegExp(`^${newPath}$`, "g");
+    this.oasSchema.paths[oldPath][UNMOCK_PATH_REGEX_KW] = newPathRegex;
+  }
+
+  private __updateSchemaPaths() {
+    Object.keys(this.schema.paths).forEach((path: string) => {
```

Comment:
Quite a massive thing inside a `forEach`, could this be somehow split into smaller pieces or include a docstring?

Reply:
I feel like it includes many docstrings, but I'll try and break it down further anyway. Thanks!